src/FESpaces/FESpacesWithDofPotentiallyRemoved.jl

```diff
@@ -0,0 +1,185 @@
+"""
+    struct FESpaceWithDofPotentiallyRemoved{CS,RD} <: SingleFieldFESpace
```

Comment:
`FESpaceConstantFixed` or  `FESpaceWithConstantFixed`

Note the difference between having the constants fixed and zero mean value

Reply:
I will use `FESpaceWithConstantFixed`. The potentially sense of the type is lost, but we can leave with that.

> Note the difference between having the constants fixed and zero mean value

Ok, good point. This space lets you fix the constant to an arbitrary value. The zero mean space fixes it to a particular value, i..e, the one which results in a zero mean function.

Reply:
> This space lets you fix the constant to an arbitrary value.

FYI, the zero mean FE space currently assumes that the value is fixed exactly to 0 in the fixed dof of the `FESpaceWithDofPotentiallyRemoved`.

Reply:
>FYI, the zero mean FE space currently assumes that the value is fixed exactly to 0 in the fixed dof of the FESpaceWithDofPotentiallyRemoved.

I do not see this statement. Doesn't it depend on the dirichlet values that come along with the trial FE space?

Reply:
Done in faa32a397b4fdb261612c37fad0d73a2e5ce2658